### PR TITLE
fix string class labels issue with mimic explainer

### PR DESCRIPTION
**Pull request by imatiach-msft**
Fix string class labels issue with mimic explainer

Specifically, for some classifiers like SGDClassifier that don't have a predict_proba method and don't output probabilities, we were failing on using string labels.  The best solution is in the model_wrapper, where we can wrap the model to our specifications to output probabilities as a boolean vector array where class i = 1 and all else are 0.  This fix ensures that any classification models without a predict_proba (we will need to whitelist them) can fit our specifications for blackbox explainers.

Resolves user issue:
https://github.com/MicrosoftDocs/azure-docs/issues/59129#event-3571829770

### CODE REVIEW

File: python/interpret_community/common/model_wrapper.py [hunk 1]
```diff
@@ -231,6 +232,45 @@ def predict(self, dataset):
...
+        :param dataset: The dataset to predict_proba on.
+        :type dataset: DatasetWrapper
+        """
+        predictions = self.predict(dataset)
+        # Generate trivial boolean array for predictions
+        probabilities = np.zeros((predictions.shape[0], self._num_classes))
+        for row_idx, pred_class in enumerate(predictions):
+            class_index = self._classes_to_index[pred_class]
+            probabilities[row_idx, class_index] = 1
+        return probabilities
```
Comment (gregorybchris) [inline python/interpret_community/common/model_wrapper.py:271]: One alternative method is to calibrate (possibly [with sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.calibration.CalibratedClassifierCV.html)) so that you always get probabilities. Do we want a different base class that uses this method?

File: python/interpret_community/mimic/model_distill.py [hunk 2]
```diff
@@ -67,6 +67,7 @@ def _model_distill(teacher_model_predict_fn, uninitialized_surrogate_model, data
                                                         **explainable_model_args)
     else:
         surrogate_model = uninitialized_surrogate_model(**explainable_model_args)
+    import pdb; pdb.set_trace()
```
Comment (gregorybchris) [inline python/interpret_community/mimic/model_distill.py:70]: Do you mean to leave this in?

File: python/interpret_community/mimic/model_distill.py [hunk 3]
```diff
@@ -76,5 +77,6 @@ def _model_distill(teacher_model_predict_fn, uninitialized_surrogate_model, data
         # Use hard labels for regression or multiclass case
         if training_labels is None:
             training_labels = teacher_y
+        import pdb; pdb.set_trace()
```
Comment (gaugup) [inline python/interpret_community/mimic/model_distill.py:80]: Same as above to what @gregorybchris alluded to for set_trace()

File: python/interpret_community/mimic/model_distill.py (hunk 2 above)
Comment (imatiach-msft) [reply python/interpret_community/mimic/model_distill.py:70]: yes this should be removed, good catch!

File: python/interpret_community/mimic/model_distill.py (hunk 3 above)
Comment (imatiach-msft) [reply python/interpret_community/mimic/model_distill.py:80]: yes this should be removed, good catch!

File: test/test_mimic_explainer.py [hunk 4]
```diff
@@ -339,6 +346,46 @@ def test_explain_model_imbalanced_classes(self, mimic_explainer):
...
+        ])
+        cat_pipe = Pipeline([
+            ('cat_imputer', SimpleImputer(strategy='constant', fill_value='?')),
+            ('cat_encoder', OneHotEncoder(handle_unknown='ignore', sparse=False))
+        ])
+        feat_pipe = ColumnTransformer([
+            ('num_pipe', num_pipe, pipe_cfg['num_cols']),
+            ('cat_pipe', cat_pipe, pipe_cfg['cat_cols'])
+        ])
+        X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, test_size=0.1)
```
Comment (gaugup) [inline test/test_mimic_explainer.py:371]: X_test, y_test are never used below. Why split the dataset and do other operations on them?

Comment (gaugup) [reply test/test_mimic_explainer.py:371]: Don't you need X_test for transform as well.

Comment (imatiach-msft) [reply test/test_mimic_explainer.py:371]: good point -- actually most of this code came directly from the user with minor modifications, I didn't realize some of the variables were not used

Comment (imatiach-msft) [reply test/test_mimic_explainer.py]: removed

**Comment by gaugup** [comment]
Looks good. Approved.

File: python/interpret_community/common/model_wrapper.py (hunk 1 above)
Comment (imatiach-msft) [reply python/interpret_community/common/model_wrapper.py:271]: Agree, had this thought as well, but I think for now I wanted to do the simpler thing.  The user can add calibration on their own for now by wrapping their model and adding predict_proba.

Comment (imatiach-msft) [reply python/interpret_community/common/model_wrapper.py:271]: It might be good to add an additional helper class though that the user could eventually use.
